src/heka/sandbox.c

```diff
@@ -375,30 +379,36 @@ static void set_restrictions(lua_State *lua, lsb_heka_sandbox *hsb)
   }
   lua_pop(lua, 1); // remove max_message_size
 
-  lua_getfield(lua, 1, LSB_LOGGER);
-  size_t len;
-  const char *tmp = lua_tolstring(lua, -1, &len);
-  if (tmp) {
-    hsb->name = malloc(len + 1);
-    if (hsb->name) strcpy(hsb->name, tmp);
-  }
-  lua_pop(lua, 1); // remove the Logger
+  lua_getfield(lua, 1, LSB_OVERRIDE_HEADERS);
+  const int override_headers = lua_toboolean(lua, -1);
+  lua_pop(lua, 1); // remove the override_headers boolean
+
+  if (!override_headers) {
+    lua_getfield(lua, 1, LSB_LOGGER);
+    size_t len;
+    const char *tmp = lua_tolstring(lua, -1, &len);
+    if (tmp) {
+      hsb->name = malloc(len + 1);
```

Comment:
Sorry I lead you astray with my earlier comment.  Hostname is only used by the header update, name is used in many other places (for logging) and should not be null. We could use hostname as a flag but having another flag like you proposed is clearer.